### ISSUE TITLE
coded character output

### ISSUE THREAD

**Issue by smsnery** [rtravis/SkypeCacheViewer#9]
Hi rtravis.
First of all, congratulations for the program.
I'm using it, but the output data seems a little bit confused.
What is the coded character (charset) of output data? UTF-8? some kind of ANSI, other thing else?

Thanks
Silvio.

**Comment by rtravis** [comment]
Hi, I'm glad you like it.

The output of the program should be encoded using the UTF-8 encoding. Also the messages use some kind of mark-up language to include text formatting instructions.

**Comment by smsnery** [comment]
Thanks for your answer rtRavis. Looking at the output messages, I see some messages in english language using the UTF-8 encoding and others messages in a latin language with special characters as "ç", "ã" and so on. 
However, when a open the output file into text editors in UTF-8 (e.g. notepad++), those especial characters aren't being correctly show. If I change the enconding of text editor to some kind of ANSI, the special characters will be correctly show, but others characters in english language (e.g. "it's") aren't correctly show anymore.
And about the mark-up language, I didn't see any mark-up languages in the messages that I reported above.

What can I do to solve this problem?

Thanks again!

**Comment by rtravis** [comment]
Can you please try again this scenario with the latest source changes from the "master" branch? If the issue is still present can you please send me the binary representation of a message for which the special characters decoding fails?

You should be able to save the message like this:

```C++
#include <fstream>

static bool parse_skype_message_blob(const uint8_t *data, size_t size)
{
    // ...
    std::ofstream ofs("message.bin", std::ios_base::binary);
    ofs.write(data, size);
    // ...
}
```
after that the file "message.bin" should contain the message that needs to be parsed.

**Comment by rtravis** [comment]
This should be fixed in commit: 149c4be6ba5. After all some strings were encoded as "ISO-8859-1" and needed conversion to "UTF-8".
